qualtran/bloqs/basic_gates/global_phase.py

```diff
@@ -58,6 +77,27 @@ def my_tensors(
 
         return [qtn.Tensor(data=self.coefficient, inds=[], tags=[str(self)])]
 
+    def get_ctrl_system(
+        self, ctrl_spec: Optional['CtrlSpec'] = None
+    ) -> Tuple['Bloq', 'AddControlledT']:
+
+        # Delegate to superclass logic for more than one control.
+        if not (ctrl_spec is None or ctrl_spec == CtrlSpec()):
+            return super().get_ctrl_system(ctrl_spec=ctrl_spec)
+
+        # Otherwise, it's a ZPowGate
+        exponent: float = np.real_if_close(np.log(self.coefficient) / (np.pi * 1.0j)).item()  # type: ignore
+        bloq = ZPowGate(exponent=exponent, eps=self.eps)
```

Comment:
Please handle the case when `self.coefficient` is symbolic

Reply:
Should we also have an override where we return a `ZGate` instead of `ZPowGate` if `exponent` is `1` ?

Reply:
That's a good question, and I think it's part of a broader issue. How do we handle cases where you can express the same "thing" (unitary) via different bloq+attribute combinations? I think the answer is to keep things as non-magical as possible when constructing bloqs and handle overlap in the bloq analysis functions. Happy to discuss more.

For this PR in particular: since `ZPowGate(exponent=1)` is a valid bloq, I'll leave as-is. 

I'll handle symbolic coefficient